### DISCUSSION TITLE
Get files without SFTP

### DISCUSSION THREAD

**Discussion by felipenaf** [phpseclib/phpseclib#1809] [Q&A] [answered]
Hi, everyone.

I'm trying to transfer files, but the remote server doesn't have an SFTP installed. 

It's possible to transfer files without the SFTP class?
Use just SCP?
Or another way, I don't know.

In any case, the error that I received is this:

```
UnexpectedValueException: Expected NET_SFTP_VERSION. Got packet type: 
/home/felipe/dev/vam5.6/be/API/composer/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php:612
/home/felipe/dev/vam5.6/be/API/composer/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php:637
/home/felipe/dev/vam5.6/be/API/composer/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php:532
/home/felipe/dev/vam5.6/be/API/composer/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php:1063
/home/felipe/dev/vam5.6/be/API/composer/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php:984
/home/felipe/dev/vam5.6/be/API/composer/vendor/phpseclib/phpseclib/phpseclib/Net/SFTP.php:970
/home/felipe/dev/vam5.6/be/API/tests/SFTPTest.php:62
```

I'm using version 3.0

I appreciate the help.

Thanks!

**Comment by terrafrost** [comment, marked as answer]
Per https://unix.stackexchange.com/a/282220/54354 you should be able to do `$ssh->exec('hexdump -C yourfile.bin')` or `$ssh->exec('xxd -b yourfile.bin`.

I dropped SCP from phpseclib v3 because (1) I think SFTP is the superior protocol and (2) #146 defied diagnosis.

**Reply by felipenaf** [reply]
I was trying to find some answer to this, but I didn't, until now.
@terrafrost, thank you very much for the explanation. 
And I agree about the SFTP. In any other case, I will adopt him even.
